crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -59,16 +60,14 @@ fn process_abi_with_input(
     } else {
         0
     };
-    for (param_name, param_type) in abi.parameters.into_iter() {
+    for (param_name, param_type) in abi.parameters.clone().into_iter() {
         let value = witness_map
```

Comment:
this check could also be in a Abi function check_param(param_name,type, witness_map);